src/plugin/morphologyHDF5.cpp

```diff
@@ -213,35 +213,53 @@ void MorphologyHDF5::_readPoints(unsigned int firstSectionOffset)
                 "Reading morphology file '" + _file->getName() + "': bad number of dimensions in 'points' dataspace"));
         }
         std::vector<std::vector<float>> vec(dims[0]);
-        // points.resize(dims[0]);
         dataset.read(vec);
-        bool noNeurites = (firstSectionOffset == -1);
-        for (unsigned int i = 0; i < vec.size(); ++i) {
+
+        if (firstSectionOffset < 0) {
+            firstSectionOffset = vec.size();
```

Comment:
I don't know what's going on here, but maybe it's better to introduce a new variable if we change data type to unsigned here (perhaps it implies the change of the semantics as well?).

Reply:
I don't think we should worry about that just yet. Besides, do `std::size_t` and `unsigned int` differ?

Reply:
`firstSectionOffset` is an `int` but is used as `unsigned int` from this point (also compared with `unsigned int` in the subsequent `for`).
It might be that it's worth reflecting this "type change" with an intermediate variable.
I haven't looked into `firstSectionOffset < 0` semantics though.

Reply:
> I don't think we should worry about that just yet. Besides, do `std::size_t` and `unsigned int` differ?

Yes, at least w/ clang 6: https://godbolt.org/z/DU8wad